src/tree/index.rs

```diff
@@ -85,13 +85,13 @@ impl Index<LeafIndex> for Vec<Node> {
     /// same leaf, but in the array representing the tree as opposed to the one
     /// only containing the leaves.
     fn index(&self, leaf_index: LeafIndex) -> &Self::Output {
-        &self[leaf_index.as_usize() * 2]
+        &self[NodeIndex::from(leaf_index).as_usize()]
```

Comment:
Ha so this was broken. `as_usize` does the `*2` already. The result doesn't look nice though. Let's clean that up in a follow up.